include/usb/usb_device.h

```diff
@@ -38,11 +38,18 @@
 
 #include <drivers/usb/usb_dc.h>
 #include <usb/usbstruct.h>
+#include <logging/log.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#define USB_DBG(fmt, ...) LOG_DBG("(%p): %s: " fmt, k_current_get(), \
```

Comment:
I don't get those wrappers. Imo, it would be better to create dedicated macro for logging current function and thread and use that at the beginning of functions (instead of USB_DBG("") which is confusing) and use LOG_... elsewhere. I can imagine that registers dump output must look ugly with all registers prefixed with function name and a pointer.

Reply:
> I don't get those wrappers.

In networking side we are using similar macros like NET_DBG() etc. This was a lifesaver as when moving from syslog -> new logger, it made life so much easier as only the macro definition needed to change. It is totally different thing what the macro actually prints and if all extra fields are always needed.

Reply:
@nordic-krch  At the moment simple LOG_ is not very useful since it does not print function name. Without function names most logs are unreadable.
BTW: I think that register dump should come to logs at all.

Reply:
USB_* macros were a request from the maintainer of usb subsys, the same way networking subsys does.

Reply:
@finikorg log message is prefixed with module name (in most cases that is almost file name). Is it needed to add __function__ to all log messages on DBG level then? Why add this to DBG and not to INF then? Maybe it would be better to have macros like USB_LOG_FUNC_ENTER, USB_LOG_FUNC_RETURN then it can be defined to anything you want (e.g. LOG_DBG(__func__, __line__)).

Reply:
@ydamigos regarding USB_* macros. If that's the requested way then i'm ok with that :)

Reply:
@nordic-krch I think the best approach would be to add configuration option to Kconfig adding function name to all LOG_*. Then for LOG_DBG and LOG_INF it should be set by default.
We use similar size for logger name so simple grep could help us to change logs later on.

Reply:
> I think the best approach would be to add configuration option to Kconfig adding function name to all LOG_*. Then for LOG_DBG and LOG_INF it should be set by default.

That would be good and more better as default for all levels.
Function name in log output is very helpful during testing or debugging. I wonder why this behavior of SYS_LOG was not taken over in new logging system.

Reply:
In networking debug prints, the thread id is also printed. This helps to differentiate the debug prints from each other because some prints can be printed with multiple threads.

Reply:
> USB_* macros were a request from the maintainer of usb subsys, the same way networking subsys does.

just because net and bluetooth went their own way (before sys_log was introduced) does not mean we have to keep "forking" the implementation and coming up with new macros doing the same thing only for a subset of the subsystems that are being logged.

If this is the only reason to have NET_DBG, BT_DBG and USB_DBG, I would say let us implement this as a feature of the logger and have it done in a consistent way across the code base.

Reply:
@nashif Our logging macros have same size so this should be simple grep for changing them. At he moment we cannot use logger as is.